src/Meshes/trailMesh.ts

```diff
@@ -0,0 +1,156 @@
+
+import { AbstractMesh } from "./abstractMesh";
+import { Mesh } from "./mesh";
+import { Material } from "../Materials/material";
+import { VertexBuffer } from "./buffer";
+import { VertexData } from "./mesh.vertexData";
+import { Vector3 } from "../Maths/math";
+import { Scene } from "../scene";
+
+/**
+ * Class used to create a trail following a mesh
+ */
+export class TrailMesh extends Mesh {
+	private _generator: AbstractMesh;
+	private _diameter: number;
+	private _length: number;
+	private _sectionPolygonPointsCount: number = 4;
+	private _sectionVectors: Array<Vector3>;
+	private _sectionNormalVectors: Array<Vector3>;
+
+    /**
+     * @constructor
+     * @param name The value used by scene.getMeshByName() to do a lookup.
+     * @param generator The mesh to generate a trail.
+     * @param scene The scene to add this mesh to.
+     * @param diameter Diameter of trailing mesh. Default is 1.
+     * @param length Length of trailing mesh. Default is 60.
+     * @param material Material to apply to trailing mesh. Defaults to scene default.
+     */
+	constructor(name: string, generator: AbstractMesh, scene: Scene, diameter: number = 1, length: number = 60, material: Material) {
+		super(name, scene);
+		scene = this.getScene();
+
+		this.layerMask = 2;

```

Comment:
why this?

Reply:
Sorry, I saw that somewhere else in another file. Wasn't sure if it was a common thing, I'll get rid of it.